tools/gulp/tasks/docs.ts

```diff
@@ -37,3 +40,32 @@ task('api', () => {
   const dgeni = new Dgeni([docsPackage]);
   return dgeni.generate();
 });
+
+/** Updates the markdown file's content to work inside of the docs app. */
+function transformMarkdownFiles(buffer: Buffer, file: any): string {
+  let content = buffer.toString('utf-8');
+
+  /* Replace <!-- example(..) --> comments with HTML elements. */
+  content = content.replace(EXAMPLE_PATTERN, (match: string, name: string) =>
+    `<div material-docs-example="${name}"></div>`
+  );
+
+  content = content.replace(LINK_PATTERN, (match: string, pre: string, link: string) =>
```

Comment:
Can you add comments for what `match`, `pre`, and `link` are expected to contain? (just an example would be good)